compiler/rustc_expand/src/mbe/transcribe.rs

```diff
@@ -752,39 +763,37 @@ fn transcribe_metavar_expr<'a>(
 }
 
 /// Extracts an metavariable symbol that can be an identifier, a token tree or a literal.
-fn extract_var_symbol<'a>(
+fn extract_symbol_from_pnr<'a>(
     dcx: DiagCtxtHandle<'a>,
     ident: Ident,
-    interp: &FxHashMap<MacroRulesNormalizedIdent, NamedMatch>,
+    pnr: &ParseNtResult,
 ) -> PResult<'a, Symbol> {
```

Comment:
Could you make all this a `match`?

Reply:
Refactored to use `match` but it is important to note that the only diff in this function is the removal of the outer most block.